content/post/2020-06-01-macos-installer/index.Rmd

```diff
@@ -0,0 +1,265 @@
+---
+slug: macos-installer
+title: "Using muliple versions of R on macOS"
+authors:
+  - Gábor Csárdi
+date: "2020-06-01"
+tags:
+- package development
+- installer
+- R versions
+output: hugodown::hugo_document
+---
+
+<!-- -*- mode: gfm; canonical: true -*- -->
+
+```{r include = FALSE}
+knitr::knit_engines$set(bash = function(options) {
+  `%||%` <- function(l, r) if (is.null(l)) r else l
+  code <- paste0(options$code, collapse = "\n")
+  options$code <- paste("$", options$code)
+
+  filename <- file.path(
+    knitr::opts_chunk$get("fig.path"),
+    paste0(options$label, ".gif")
+  )
+  
+  if (!file.exists(filename)) {
+    tmp <- tempfile(fileext = ".cast")
+    on.exit(unlink(tmp), add = TRUE)
+
+    processx::run(
+      "asciinema",
+      c("rec", "-i", "2", "-c", code, tmp),
+      echo = TRUE
+    )
+    cast <- asciicast::read_cast(tmp)
+  
+    dir.create(
+      dirname(filename), 
+      showWarnings = FALSE, 
+      recursive = TRUE
+    )
+
+    asciicast::write_gif(
+      cast, 
+      filename, 
+      rows = options$rows %||% "auto", 
+      theme = "asciinema", loop = 1,
+      speed = options$speed %||% 1.0
+    )
+  }
+  
+  extra <- knitr::knit_hooks$get('plot')(filename, options)
+  knitr::engine_output(options, options$code, '', extra)
+})
+```
+
+## TL;DR
+
+If you need to use multiple versions of R on macOS, look at the
+`install-rstats` package in the 'Installer' section below.
+
+## Introduction
+
+As an R package developer, it is mostly your choice which R versions
+you support. If your package is on CRAN, then you must support the
+latest R release and also fix issues that come up when your package is
+tested on the current development version (R-devel). These two
+versions are the minimum, but developers typically try to do better.
+E.g. the tidyverse supports the [four previous versions or
+R](https://www.tidyverse.org/blog/2019/04/r-version-support/).
+
+The more versions you support, the more versions you typically keep on
+your development machine, to be able to reproduce problems and fix
+bugs, that are specific to older or newer versions. [R-hub is
+great](https://blog.r-hub.io/2019/03/26/why-care/) for quick checks,
+but if there is a problem, you'll want to run R on your local machine
+to investigate. In this post we show a way to run multiple R versions
+on macOS, at the same time.
+
+## The macOS R installer
+
+The official CRAN R installers work differently on different operating
+systems. On Windows, every R version is installed into its own
+directory and works independently of the others. On Linux,
+distributions typically only support one R version at any time, but
+you can easily use Docker to run [multiple R
```

Comment:
do you have any link to a good starter Docker tutorial? Or maybe a link to /2020/01/08/cran-error/#r-and-docker (https://blog.r-hub.io/2020/01/08/cran-error/#r-and-docker)